src/main/scala/edu/ie3/simona/model/control/TransformerControlGroupModel.scala

```diff
@@ -0,0 +1,213 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.control
+
+import breeze.math.Complex
+import edu.ie3.datamodel.models.input.MeasurementUnitInput
+import edu.ie3.powerflow.model.NodeData.StateData
+import edu.ie3.powerflow.model.PowerFlowResult.SuccessFullPowerFlowResult
+import edu.ie3.simona.config.SimonaConfig
+import edu.ie3.simona.config.SimonaConfig.TransformerControlGroup
+import edu.ie3.simona.model.control.TransformerControlGroupModel.RegulationCriterion
+import squants.{Dimensionless, Each}
+
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+
+/** Business logic for a transformer control group. It's main purpose is to
+  * determine, if there is any regulation need and if yes, to what extent (here:
+  * voltage raise or reduction to achieve)
+  *
+  * @param nodalRegulationCriterion
+  *   Mapping from nodal index to a partial function, that determines the
+  *   regulation need at this node
+  * @param harmonizeRegulationNeeds
+  *   Partial function to harmonize different, possible contradictory regulation
+  *   needs
+  */
+final case class TransformerControlGroupModel(
+    nodalRegulationCriterion: Map[UUID, RegulationCriterion],
+    harmonizeRegulationNeeds: Array[Dimensionless] => Option[Dimensionless],
```

Comment:
I think this parameter has now become obsolete with the introduction of `TransformerControlGroupModel.harmonizationFunction`.